code/datums/communication/donator.dm

```diff
@@ -12,9 +12,12 @@
 	if(!.)
 		return
 
-	if(!C.holder)
+	if(check_rights(R_INVESTIGATE))
+		return
+	else
 		if(!(C.donator_holder && C.donator_holder.flags & D_DOOC))
-			return FALSE
+			. = FALSE
+	return
```

Comment:
What the hell is this supposed to be?

`else` on an early return if statement? I don't get the `.` assignment either..